bolt/lib/Passes/PAuthGadgetScanner.cpp

```diff
@@ -382,11 +382,11 @@ class PacRetAnalysis
 
 public:
   std::vector<MCInstReference>
-  getLastClobberingInsts(const MCInst Ret, BinaryFunction &BF,
-                         const ArrayRef<MCPhysReg> UsedDirtyRegs) const {
```

Comment:
I was wondering whether `const` has to be removed here.
Looking at the implementation of `getStateAt` and `getStateBefore`, it seems that it might be better to add the `const` qualifier to `getStateBefore` at https://github.com/llvm/llvm-project/blob/9768077de65e31daa619eae231f027e052d601c2/bolt/include/bolt/Passes/DataflowAnalysis.h#L295?

Reply:
This sounds reasonable and seems to be important for detecting signing oracles (to handle instruction sequences inserted by the compiler to check authenticated register - this would make possible to call `getStateBefore` from `computeNext`), but it is probably better to address via a separate PR.